TrioAnalysis.pm

```diff
@@ -0,0 +1,253 @@
+
+
+=cut
+
+package TrioAnalysis;
+
+use strict;
+use warnings;
+use Cwd;
+
+use Bio::EnsEMBL::Variation::VariationFeature;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepFilterPlugin);
+
+
+sub _parse_ped_file {
+  my $self = shift;
+  my $file = shift;
+
+  open FILE, $file;
+  while(<FILE>) {
+    chomp;
+    my ($family_id, $ind_id, $paternal_id, $maternal_id, $sex, $pheno) = split/\s+|\t/;
+    
+    if(!defined ($family_id && $ind_id && $paternal_id && $maternal_id && $sex && $pheno)) {
+      die "ERROR: PED file requires dix columns: family ID, individual ID, paternal ID, maternal ID, sex, phenotype\n";
+    }
+    
+    $self->{linkage}->{$ind_id}->{paternal_id} = $paternal_id;
+    $self->{linkage}->{$ind_id}->{maternal_id} = $maternal_id;
+    $self->{linkage}->{$ind_id}->{sex} = $sex;
+    $self->{linkage}->{$ind_id}->{pheno} = $pheno;
+    $self->{linkage}->{$ind_id}->{child} = 1 if ($pheno == 2);
+  }
+  close FILE;
+}
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+
+  my $param_hash = $self->params_to_hash();
+
+  if (defined $param_hash->{ped} && -e $param_hash->{ped}) {
+    $self->_parse_ped_file($param_hash->{ped});
+  }
+  else {
+    die "ERROR: Please provide a valid PED file\n";
+  }
+
+  if (defined $param_hash->{report_dir}) {
+    if (!-d $param_hash->{report_dir}) {
+      my $return = mkdir $param_hash->{report_dir}, 0755;
+      die("ERROR: Couldn't create report_dir ", $param_hash->{report_dir}, " $!\n") if (!$return);
+    }
+    $self->{report_dir} = $param_hash->{report_dir};
+  }
+  else {
+    my $cwd_dir = getcwd;
+    $self->{report_dir} = "$cwd_dir";
+  }
+
+  # force some config params
+  $self->{config}->{individual_zyg} = ['all'];
+
+  # Report files
+  $self->{report_de_novo} = "variants_de_novo.txt";
+  $self->{report_all} = "variants_child_and_both_parents.txt";
+  # Write header
+  write_header($self->{report_dir}.'/'.$self->{report_de_novo}, 1);
+  write_header($self->{report_dir}.'/'.$self->{report_all});
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Feature', 'Intergenic'];
+}
+
+sub get_header_info {
+  my $self = shift;
+
+  my %header;
+
+  $header{'TrioAnalysis'} = '';
+
+  return \%header;
+}
+
+sub run {
+  my ($self, $tva, $line) = @_;
+  
+  my $vf = $tva->variation_feature;
+  my $zyg = defined($line->{Extra}) ? $line->{Extra}->{ZYG} : $line->{ZYG};
+  
+  my $chr = $vf->{chr};
+  my $start = $vf->{start};
+
+  my $result;
+  my $list_of_ind;
+
+  foreach my $geno_ind (@{$zyg}) {
+    my ($ind, $geno) = split(':', $geno_ind);
+
+    # Check if VCF and PED file have the same individual IDs
+    if(!$self->{linkage}->{$ind}) {
+      die "ERROR: VCF and PED individuals do not match. Please check the individual IDs in PED file: ", join(',', keys %{$self->{linkage}}), "\n";
+    }
+
+    # HOM : homozygous
+    # HET : heterozygous
+    # HOMREF : homozygous reference (not a variant)
+    if($geno eq 'HOM' || $geno eq 'HET') {
```

Comment:
If the parents are homozygous alt and the child heterozygous, we should report this.

Can we compare probands to their parents here and support many rather than using the child flag?

Reply:
I updated it to support multiple families.